### ISSUE TITLE
fix require for clashing structs/defs 

### ISSUE THREAD

**Issue by vitreo12** [vitreo12/omni#98]
`One.omni` :

```nim
struct Something:
    a

def someFunc():
    return 0
```

`Two.omni` :

```nim
struct Something:
    a; b

def someFunc():
    return 1
```

`Three.omni`:

```nim
require "One.omni", "Two.omni"

init:
    one = One.Something()
    two = Two.Something()

sample:
    out1 = One.someFunc() + Two.someFunc()
```

**Comment by vitreo12** [comment]
One thing to highlight is that probably this would require a way to access / set fields of a `struct`, without making them explicitlty public. One solution is to overload `method` like this:

```nim
type
    Euler* = object
        x, y, z, a, b, c: float64

# ... but expose methods like these
method `x`*(this: Euler): float64 =
    return this.x

method `x=`*(this: Euler, x:float64) =
    this.x = x
```

**Comment by vitreo12** [comment]
Some progress here: #102

**Comment by vitreo12** [comment]
Main problem now is how using `structs` with same name as imported module

**Comment by vitreo12** [comment]
New PR at #115

**Comment by vitreo12** [comment]
Results of new PR:
`One.omni:`

```nim
struct Something:
    a

def someFunc():
    return 0.7
```

`Two.omni`

```nim
struct Something:
    a

def someFunc():
    return 0.3
```

`Three.omni:`

```nim
use One:
    Something as Something1
    someFunc as someFunc1

use Two:
    Something as Something2
    someFunc as someFunc2

init:
    one = Something1()
    two = Something2()

sample:
    out1 = someFunc1() + someFunc2()
```

**Comment by vitreo12** [comment]
Merged! #117
